src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -745,7 +753,13 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                 disposer = null;
 
                 // Have to add 1 to iinfo since the enum starts at 1
-                return Utils.MarshalInvoke(MakeGetter<int>, ((DateTimeEstimator.ColumnsProduced)iinfo + 1).GetRawColumnType(), input, iinfo);
+                // If input type is long, Posix time
+                if (input.Schema[_parent._column.Source].Type.RawType == typeof(long))
+                    return Utils.MarshalInvoke(MakeGetter<int, int>, new Type[] { typeof(long), ((DateTimeEstimator.ColumnsProduced)iinfo + 1).GetRawColumnType() }, input, iinfo);
+                // If input types is System.DateTime
+                else
```

Comment:
Would an assert be better than a comment?

Reply:
I was just trying to document the code execution path. If the type is not a supported type then an error will be thrown before it gets to this point.